sbc_switzerland/models/correspondence.py

```diff
@@ -122,14 +122,11 @@ def process_letter(self):
         intro_letter = self.env.ref(
             'sbc_compassion.correspondence_type_new_sponsor')
         for letter in self:
-            # if letter.original_language_id in \
-            #         letter.supporter_languages_ids or \
-            if (letter.beneficiary_language_ids &
-                    letter.supporter_languages_ids) or \
+            if intro_letter in letter.communication_type_ids and not \
+                    letter.sponsorship_id.send_introduction_letter:
```

Comment:
Couldn't we just check only _letter.sponsorship_id.send_introduction_letter_ ?

Reply:
I don't think so, as all letters pass through this function, is is only `intro_letters` where `send_introduction_letter` is false that must be skipped. Otherwise this would avoid sending all letters of sponsorships where `send_introduction_letter` is set to false.